### DISCUSSION TITLE
v0.8.7

### DISCUSSION THREAD

**Discussion by vfsfitvnm** [vfsfitvnm/frida-il2cpp-bridge#390] [Announcements]
## Relevant changes

- Add  `IL2CPP_MODULE_NAME`: \
  A variable to override the default module name where IL2CPP exports should be searched into. It's for advanced use cases. 
  ```ts
  (gloablThis as any).IL2CPP_MODULE_NAME = "whatever";

  Il2Cpp.perform(() => {
      // ...
  });
  ```

- Add  `IL2CPP_UNITY_VERSION`: \
  A variable to set/override the Unity version for when it cannot be detected automatically. It's for advanced use cases.
  ```ts
  (gloablThis as any).IL2CPP_UNITY_VERSION = "2019.2.3f1";

  Il2Cpp.perform(() => {
      console.log(IL2CPP_UNITY_VERSION); // 2019.2.3f1
  });
  ```

- Add  `IL2CPP_EXPORTS`: \
  A variable to specify IL2CPP exports providers. It's for advanced use cases, for instance when some IL2CPP exports are stripped/hidden/renamed and cannot be detected automatically.
  ```ts
  declare global {
      let IL2CPP_EXPORTS: Record<string, () => NativePointer>;
  }
  
  IL2CPP_EXPORTS = {
      il2cpp_image_get_class: () => Il2Cpp.module.base.add(0x1204c),
      il2cpp_class_get_parent: () => {
          return Memory.scanSync(Il2Cpp.module.base, Il2Cpp.module.size, "2f 10 ee 10 34 a8")[0].address;
      },
  };

  Il2Cpp.perform(() => {
      // ...
  });
  ```

- Improve and fix some annoyances related to native modules hooking.

**Full Changelog**: https://github.com/vfsfitvnm/frida-il2cpp-bridge/compare/v0.8.6...v0.8.7

**Comment by commonuserlol** [comment]
> **Full Changelog**: [v0.8.2...v0.8.3](https://github.com/vfsfitvnm/frida-il2cpp-bridge/compare/v0.8.2...v0.8.3)

Wrong changelog
